### PR TITLE
feat(ui): Implement Unfied view standard filters

### PR DESCRIPTION
**Pull request by bdauria**
# Pull Request Template

## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [x] I have made sure that the **unit tests** related to the story are successful.
- [x] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/front_src/src/Resources/Filter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  };
+
+  const requestSearch = (): void => {
+    onSearchRequest(searchFieldValue);
+  };
+
+  const getHostGroupSearchEndpoint = (searchValue): string => {
+    return buildHostGroupsEndpoint({
+      limit: 10,
+      search: `name:^${searchValue}`,
```
Comment (kduret) [inline www/front_src/src/Resources/Filter/index.tsx]: I'm not sure we want to search Only hostgroups which begin by search value. 
We should remove `^`

File: www/front_src/src/Resources/Filter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    });
+  };
+
+  const getServiceGroupSearchEndpoint = (searchValue): string => {
+    return buildServiceGroupsEndpoint({
+      limit: 10,
+      search: `name:^${searchValue}`,
```
Comment (kduret) [inline www/front_src/src/Resources/Filter/index.tsx]: Same here

File: www/front_src/src/Resources/columns/index.tsx [hunk 3]
```diff
@@ -76,7 +76,7 @@ const ResourcesColumn = ({ Cell, row }: ColumnProps): JSX.Element => {
           <Grid container spacing={1}>
             <Grid item xs={1} />
             <Grid item>
-              <StatusChip statusCode={row.parent.status.code} />
+              <StatusChip statusCode={row.parent?.status?.code || 0} />
```
Comment (kduret) [inline www/front_src/src/Resources/columns/index.tsx]: Default status code may be pending

File: www/front_src/src/Resources/index.test.tsx [hunk 4]
```diff
@@ -122,33 +174,122 @@ describe(Resources, () => {
...
+      optionToSelect,
+      endpointParamChanged,
+      selectEndpointMockAction,
+    }) => {
+      it(`executes a listing request with selected "${filterName}" filter options when it's changed`, async () => {
+        const { getAllByText } = render(<Resources />);
+
+        await wait(() => expect(mockedAxios.get).toHaveBeenCalled());
+
+        selectEndpointMockAction?.();
```
Comment (kduret) [inline www/front_src/src/Resources/index.test.tsx]: Did not know this was possible with optional chaining :)

File: www/front_src/src/Resources/columns/index.tsx (hunk 3 above)
Comment (bdauria) [reply www/front_src/src/Resources/columns/index.tsx]: Actually this chick should be removed as the API contracts defines parent -> status -> code as mandatory whenever parent is defined

**Pull request merged**